waku/v2/node/discv5/waku_discv5.nim

```diff
@@ -63,12 +93,14 @@ proc findRandomPeers*(wakuDiscv5: WakuDiscoveryV5): Future[Result[seq[RemotePeer
   ## Find random peers to connect to using Discovery v5
   
   ## Query for a random target and collect all discovered nodes
-  ## @TODO: we could filter nodes here
   let discoveredNodes = await wakuDiscv5.protocol.queryRandom()
   
+  ## Filter based on our needs
+  let filteredNodes = discoveredNodes.filter(isWakuNode) # Currently only a single predicate
```

Comment:
Does this mean Discovery v5 logic is stopped for non-filter nodes? Trying to understand how we'll keep the network cleanly separated in case of happy little accident

Reply:
Are you referring to keeping the networks separated from the eth2 network(s) (and others)?

If so, no, this is not what it does. It only filters the nodes it just found through a query (querying a bunch of other nodes for more nodes). The underlying routing table will not get filtered and contains all sort of nodes.

Reply:
Yeah, this only uses the flag to filter the discovered nodes for usefulness elsewhere. If we want to limit what goes into the routing table we'll have to modify the underlying routing table logic (likely forking the implementation/creating our own).